teamscale_precommit_client/precommit_client.py

```diff
@@ -26,7 +28,8 @@ def __init__(self, teamscale_config, repository_path, analyzed_file=None, verify
                                                 verify)
         self.analyzed_file = analyzed_file
 
-    def upload_precommit_data(self):
+    def upload_precommit_data(self, include_existing_findings=False,
```

Comment:
How about extracting the calculation of changed and deleted files into the run method? We do not need to do that in this method. I think that would make things overall easier to read, as we have all the logic that decides on which findings to report in the run method, and we do not need any parameters and any return value here any more.

Reply:
Sure. We do need to pass `changed_files` and `deleted_files` as parameters now, but you're right, having them as return value is a bit confusing.

Reply:
`run` method is quite long now, but hopefully 36 lines is still within limits :)